Documentation update: other (#732)
flame-engine/flame: doc/debug.md

## FPS counter

> Flame provides the `FPSCounter` mixin for recording the fps, this mixin can be applied on any class

Reviewer: ```suggestion
Flame provides the `FPSCounter` mixin for recording the fps; this mixin can be applied on any class
```

that extends from `Game`. Once applied you can access the current fps by using the `fps` method,
like shown in the example below.

```dart
class MyGame extends Game with FPSCounter {
  static final fpsTextConfig = TextConfig(color: BasicPalette.white.color);

  @override
  void render(Canvas canvas) {
    super.render(canvas);

>     fpsTextConfig.render(canvas, fps(120).toString(), Vector2(0, 50));

Reviewer: probably we should document what 120 is here. I suggest saving to a variable:
```dart
final fpsCount = fps(120); // explain here
fpsTextConfig. ....  (this is only about rendering)
```

  }
}
```

## BaseGame features

> On `BaseGame` you don't have to apply the `FPSCounter` mixin to get access to the current fps,
> because it is already applied by default on the `BaseGame` class, so you can use the `fps` method directly.

Reviewer: ```suggestion
If you are using `BaseGame`, you don't have to apply the `FPSCounter` mixin to get access to the current FPS,
because it is already applied by default (so you can use the `fps` method directly).
```

> Flame provides some debugging features for the `BaseGame` class, these features are enabled when
> the `debugMode` property is set to `true` (or overridden to be `true`).

Reviewer: ```suggestion
Flame provides some debugging features for the `BaseGame` class. These features are enabled when
the `debugMode` property is set to `true` (or overridden to be `true`).
```

> When `debugMode` is enabled each `PositionComponent` will be rendered with their bounding size, and have
> their positions rendered on the screen, so you can visually verify the components boundaries and
> positions.

Reviewer: ```suggestion
When `debugMode` is enabled, each `PositionComponent` will be rendered with their bounding size, and have
their positions written on the screen. This way, you can visually verify the components boundaries and
positions.
```
